Some shuttes rework. Allow many shuttes of one template to work at the same time, and returned 2 fighter to syndicate ambush.

Dennok opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Make docking_port gen id and name.
This allow creating and working of many shuttles created by single template.

Add to shuttle initialisation behaveor to auto modification doors, cameras, camera consoles, buttons, pod doors id's to unicue per shuttle. Now Buttons from one shuttle dont open doors from another. And you cant see cameras from another copies of the ship.

Now ID tru unical, replaced part of it behaveor by destynation_type. You can set many docks of one destynation_type but all it have different id's. you can send shuttle to docks ID, Or can send shuttle to destynation_type, then it be send to first port with matches destynation_type.
Now shuttle console collect list of ID's by matching console(shuttle) and dock destynation_type. And you can send shutle to dock ID.

Previous console collect list of ID's by matching console(shuttle) and dock ID. And if list have matching ID's you you can send only to first match ID in SSshuttle.shuttles list.

And if ports have match name or id it can be dont show in some menu.

SSshuttle stationary and mobile now associated lists(id = shuttle/dock).
## Why It's Good For The Game
Now we dont need code new areas, consoles, docking ports to create new shuttles from single template.
If you need second whiteship, just place it.
If you want new docking port just place it, no need to VV it id, and VV shuttle console destinations. Just make sure that it destynation_type same, and it fit in width and height.
For whiteship's Deep Space just place whiteship dock template.

## Changelog
:cl:
add: Make many single template shuttles work at the same time.
add: Returned 2 fighter to syndicate ambush.
/:cl:

TODO:

- [x] Make `obj/docking_port` id unical, replace part of it behavior by `destination_type`
- [x] Replace all starting 'id' to `destination_type` in `obj/docking_port`
- [x] Make every shuttle have its own button codes, and cameras.
- [x] Change using shuttles.len in id to something another, to dont get messeg after deliting shuttles/docks.
- [x] Make map_template_load() propertly differentise machinery codes when load shuttles.
- [x] Make SSshuttle.mobile and SSshuttle.stationary asociative list's.
- [x] Sure that all shuttle related mechanics (what i find and test) work as intended.
- [ ] Tests
- [ ] Sure that shuttle related mechanics work as intended.
- [ ] Live tests
- [ ] Sure that absolutely all shuttle related mechanics work as intended.
- [ ] Change `destination_type` var name to some more intuitive and universal.
- [ ] Give shuttle manipulator variant to load addition shuttles, instead of replacing.
- [ ] Make shuttle manipulator less laggy.

Dennok committed:
return 2 fighter to caravan ambush

Dennok committed:
Make docking_port gen id and name

ninjanomnom commented:
The issue with multiple shuttles with one template isn't the docking port, the issue is with all shuttles made from one template sharing areas. We need changes in map loading in order to support that

ninjanomnom commented on code/modules/shuttle/shuttle.dm:
```diff
@@ -163,10 +165,8 @@
-	if(name == "dock")
-		name = "dock[SSshuttle.stationary.len]"
+	id = "[SSshuttle.stationary.len]"
+	name = "[name] dock[SSshuttle.stationary.len]"
```
Do not remove the previous code, this is to handle map edits

Dennok replied:
how to handle !00% different ID and NAME near shuttles in another way?

Dennok replied:
this code want change id if we have 2 or more of dock/shuttles but i dont see it work. so i remove the check and give unical suffix

ninjanomnom replied:
I don't understand what you're saying here

Dennok replied:
Problem in map edited shuttles, it ID and NAME same for all created by template.

ninjanomnom replied:
Then remove those map edits causing problems and being used improperly. That's not a reason to remove map edit support

Dennok replied:
I return map edit support

ninjanomnom commented on code/modules/shuttle/shuttle.dm:
```diff
@@ -290,10 +290,8 @@
-	if(name == "shuttle")
-		name = "shuttle[SSshuttle.mobile.len]"
+	id = "[SSshuttle.mobile.len]"
+	name = "[name] shuttle[SSshuttle.mobile.len]"
```
Ditto

Dennok replied:
how to handle !00% different ID and NAME near shuttles in another way?

ninjanomnom commented on code/modules/shuttle/shuttle.dm:
```diff
@@ -315,10 +313,8 @@
-		if(name == initial(name))
-			name = "[name] [idnum]"
+		id = "[id][idnum]"
+		name = "[name] [idnum]"
```
Ditto

ninjanomnom commented on code/modules/shuttle/shuttle.dm:
```diff
@@ -432,7 +428,10 @@
 		else
-			previous = S0
+			if(S0.delete_after)
+				S0.Destroy(TRUE)
```
```suggestion
				qdel(S0, TRUE)
```

Dennok committed:
Update code/modules/shuttle/shuttle.dm

Co-Authored-By: Emmett Gaines <ninjanomnom@gmail.com>

Dennok commented:
Templates actually creates different zones for spawned templates. And as i see it dont shares apc.
![scrnshot1](https://user-images.githubusercontent.com/7734424/60843828-62921d80-a1e0-11e9-98a3-2a1672d80c83.png)
Here i fix issue what from spawned from one template shuttles work only one. 
And its problem in sharing its shuttle IDs.

Dennok commented:
About ID and name
Actual: 
![image](https://user-images.githubusercontent.com/7734424/60845259-67f16700-a1e4-11e9-812f-5d8699b74e4f.png)
![image](https://user-images.githubusercontent.com/7734424/60845282-7770b000-a1e4-11e9-9c3b-61d5e23c6cbf.png)

After my changes:
![image](https://user-images.githubusercontent.com/7734424/60845725-ddaa0280-a1e5-11e9-8303-0dc8b32350cd.png)

Dennok committed:
Update buttons.dm

Make buttons id in shuttles bound to shuttle.

Dennok committed:
Update poddoor.dm

Make poddoor id in shuttles bound to shuttle.

Dennok committed:
Update airlock.dm

Make airlock id_tag in shuttles bound to shuttle.

Dennok committed:
ups

Dennok committed:
Update mass_driver.dm

Make mass driver id in shuttles bound to shuttle.

Dennok committed:
Update igniter.dm

Make igniter id in shuttles bound to shuttle.

Dennok committed:
Update camera.dm

Make camera network in shuttles bound to shuttle.

Dennok committed:
Update camera.dm

Make security camera console network in shuttles bound to shuttle.

Dennok committed:
Update camera_advanced.dm

Make advanced camera console networks in shuttles bound to shuttle,
except for syndie.

Dennok committed:
Update flasher.dm

Make flasher id in shuttles bound to shuttle.

Dennok committed:
Update morgue.dm

Make crematorium id in shuttles bound to shuttle.

Dennok committed:
ups

Dennok committed:
Update navigation_computer.dm

Dennok committed:
Update shuttle.dm

Dennok committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into caravan_return_2_fighter

Dennok committed:
up

Dennok committed:
Revert "Update shuttle.dm"

This reverts commit b2c2813edf6a1e369815380953669bcac09b1e87.

Dennok committed:
Update computer.dm

Dennok committed:
Update shuttle.dm

Dennok committed:
Update shuttle.dm

Dennok committed:
Update shuttle.dm

Dennok committed:
Update arrivals.dm

Dennok committed:
Update assault_pod.dm

Dennok committed:
Update elevator.dm

Dennok committed:
Update emergency.dm

Dennok committed:
Update supply.dm

Dennok committed:
Update ferry_base.dmm

Dennok committed:
Update ferry_fancy.dmm

Dennok committed:
Update ferry_kilo.dmm

Dennok committed:
Update ferry_lighthouse.dmm

Dennok committed:
Update ferry_meat.dmm

Dennok committed:
Update hunter_russian.dmm

Dennok committed:
Update hunter_space_cop.dmm

Dennok committed:
Update infiltrator_advanced.dmm

Dennok committed:
Update infiltrator_basic.dmm

Dennok committed:
Update labour_box.dmm

Dennok committed:
Update labour_delta.dmm

Dennok committed:
Update labour_kilo.dmm

Dennok committed:
Update mining_box.dmm

Dennok committed:
Update mining_common_meta.dmm

Dennok committed:
Update mining_delta.dmm

Dennok committed:
Update mining_kilo.dmm

Dennok committed:
Update mining_large.dmm

Dennok committed:
Update pirate_default.dmm

Dennok committed:
Update ruin_caravan_victim.dmm

Dennok committed:
Update ruin_pirate_cutter.dmm

Dennok committed:
Update ruin_syndicate_dropship.dmm

Dennok committed:
Update ruin_syndicate_fighter_shiv.dmm

Dennok committed:
Update snowdin_excavation.dmm

Dennok committed:
Update snowdin_mining.dmm

Dennok committed:
Update whiteship_box.dmm

Dennok committed:
Update whiteship_cere.dmm

Dennok committed:
Update whiteship_delta.dmm

Dennok committed:
Update whiteship_meta.dmm

Dennok committed:
Update whiteship_pubby.dmm

Dennok committed:
Update Lavaland.dmm

Dennok committed:
Update runtimestation.dmm

Dennok committed:
Update caravanambush.dmm

Dennok committed:
Update listeningstation.dmm

Dennok committed:
Update whiteshipdock.dmm

Dennok committed:
Update navigation_computer.dm

Dennok committed:
Update assault_pod.dm

Dennok committed:
Update shuttle.dm

la crutch

Dennok committed:
Merge pull request #1 from Dennok/Dennok-shuttles-patch-1

Update shuttle computer

Dennok committed:
Merge branch 'master' into many-shuttles-and-docks

ninjanomnom commented:
I'll have to check sometime but unless someone has made changes to map loading areas are shared by shuttles for sure. Shuttle movement as well doesn't account for handling existing area instances.

Dennok commented:
I think this make shuttle areas dont unique.
https://github.com/tgstation/tgstation/blob/master/code/game/area/areas/shuttles.dm#L14

Dennok committed:
up maps

Dennok committed:
up aux and cargo

Dennok committed:
add getDockedDestination_type

Dennok committed:
try to autodock

Dennok committed:
Update on_move.dm

Dennok committed:
Update aux_base.dm

Dennok committed:
up cargo

Dennok committed:
Update supply.dm

Dennok commented:
supply sells, i belive.

Dennok committed:
up

SpaceManiac commented:
@ninjanomnom #39890

ninjanomnom commented:
Oh yeah I forgot that was a thing

Dennok committed:
Merge remote-tracking branch 'origin/caravan_return_2_fighter' into many-shuttles-and-docks

Dennok committed:
Update navigation_computer.dm

Dennok committed:
Update caravanambush.dmm

Dennok committed:
Update navigation_computer.dm

docker console connects to shuttle on creation

Dennok committed:
Update navigation_computer.dm

Dennok committed:
Update computer.dm

New empty shuttle console connects to shuttle in where it build.

Dennok committed:
Update navigation_computer.dm

Dennok committed:
Merge branch 'many-shuttles-and-docks' of https://github.com/Dennok/Dennok_tgstation into many-shuttles-and-docks

Dennok committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into many-shuttles-and-docks

Dennok committed:
Update shuttles.dm

Dennok committed:
Update shuttle.dm

Dennok committed:
Update shuttle.dm

Dennok committed:
Update shuttle.dm

Dennok committed:
Update shuttle.dm

Dennok committed:
Update shuttle.dm

Dennok committed:
Update emergency.dm

Dennok committed:
make SSshuttle stationary and mobile associative list

Dennok committed:
Update shuttle.dm

Dennok committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into many-shuttles-and-docks

Dennok committed:
up

Dennok committed:
Update map_template_loadverb.dm

Dennok committed:
Update map_template_loadverb.dm

Dennok committed:
Update map_template_loadverb.dm

Dennok committed:
up

Dennok committed:
Update shuttle.dm

Dennok committed:
halt shuttle replacing

Dennok commented:
Split it on parts?

SpaceManiac commented:
I'm not sure splitting is needed. However, if it was, how would you split it? Mainly I am busy.

Dennok commented:
I can split out part of code that changes door, poddoor. buttons and etc. codes, and 2 fighter. 
But it not to much.

Dennok committed:
Shuttles rework by one commit

Dennok committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into many-shuttles-and-docks

optimumtact commented:
@ninjanomnom also need feedback from you on final state

Dennok committed:
Merge branch 'msad-rebase' of https://github.com/Dennok/Dennok_tgstation into many-shuttles-and-docks

Dennok commented:
To hard, and dirty. I separate it, and some rework.
